core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -662,15 +702,21 @@ impl<Block: BlockT> network_gossip::Validator<Block> for GossipValidator<Block>
 			let mut inner = self.inner.write();
 			inner.peers.new_peer(who.clone());
 
-			let packet = NeighborPacket {
-				round: inner.local_view.round,
-				set_id: inner.local_view.set_id,
-				commit_finalized_height: inner.local_view.last_commit.unwrap_or(Zero::zero()),
-			};
 
-			GossipMessage::<Block>::from(packet).encode()
+			inner.local_view.as_ref().map(|v| {
+				let packet = NeighborPacket {
```

Comment:
Could it be better to assign `NeighborPacket` to a `let neighor_packet = {`, and then do `if let Some(packet_data) = GossipMessage::<Block>::from(neighbor_packet).encode()`

The benefit is not holding the lock for the duration of the encoding(since it's not necessary for it).